### ISSUE TITLE
Duplicate bounds on autogenerated opam files

### ISSUE THREAD

**Issue by mseri** [ocaml/dune#3916]
## Expected Behavior
Depends on the generated opam file just contain one bound for dune:
```
depends: [
  "dune" {>= "2.7"}
  "ocaml" {>= "4.11.1"}
  "odoc" {with-doc}
]
```

## Actual Behavior

```
depends: [
  "dune" {>= "2.7" & >= "2.7"}
  "ocaml" {>= "4.11.1"}
  "odoc" {with-doc}
]
```

## Reproduction

Sorry, I cannot make the PR at the moment, I can come back to it if I can later on.

```
$ mkdir test; cd test
$ cat - > dune-project <<EOF
(lang dune 2.7)
(name test)

(generate_opam_files true)

(license Apache-2.0)
(source (github ocaml/dune))

(maintainers "Somebody")
(authors "Somebody else")

(package
 (name test)
 (synopsis "Read the description")
 (description "A thing that does things")
 (depends
  (dune (>= "2.7"))
  (ocaml (>= "4.11.1"))
  ))
EOF
$ dune build
$ cat test.opam
```

## Specifications

- Version of `dune` (output of `dune --version`): 2.7.1
- Version of `ocaml` (output of `ocamlc --version`):  4.11.1
- Operating system (distribution and version): macos catilina

## Additional information 

If the version in `(lang dune VV)` is `>=` than the bound in the package, then both bounds are appearing in the generated dune file (the bug I am reporting). Otherwise, only the largest value appears.

However, I think that when there is a mismatch between the two versions we should fail or at least emit a warning, since it is most likely a bug.

**Comment by mseri** [comment]
We have seen this recently in a number of opam-repository PRs

**Comment by jeremiedimino** [comment]
> However, I think that when there is a mismatch between the two versions we should fail or at least emit a warning, since it is most likely a bug.

I would go even further and say that we should encourage users not to write an explicit version constraint on Dune. But agreed the current behaviour is a bug.

**Comment by jeremiedimino** [comment]
My expectation is that opam accepts such constraints. So while this is a bug it's not critical. Given that, I marked this issues as "acknowledge", which means that it is something that should eventually be dealt with, but it's on nobody's stack at the moment.

**Comment by sim642** [comment]
> I would go even further and say that we should encourage users not to write an explicit version constraint on Dune. But agreed the current behaviour is a bug.

Unfortunately this is needed in cases like #4806 and the subsequent release of 2.9.1. I need to explicitly depend on `(dune (>= 2.9.1))` for my project installation to work and the project to work as installed through opam. I tried `(lang dune 2.9.1)`, but that doesn't seem to be a thing as is just read as `(lang dune 2.9)`, which would still just generate `"dune" {>= "2.9"}`, but that's too low of a bound.

**Comment by jeremiedimino** [comment]
Yes, the parsing of the `(lang dune ...)` was too loose by accident. We fixed this in 3.0. Regarding the issue you are bringing up, we generally consider that version X.Y is the latest X.Y.Z release. So if 2.9.1 is out, then `(lang dune 2.9)` correspond to Dune 2.9.1. Maybe that's too simplistic, though it has worked so far.

In any case, if Dune 2.9.0 is harmful, I think we should mark it as unavailable in opam. @kit-ty-kate is that something we can do?
